packages/office-ui-fabric-react/src/components/Nav/Nav.tsx

```diff
@@ -158,16 +158,22 @@ export class Nav extends BaseComponent<INavProps, INavState> implements INav {
     );
   }
 
-  private _renderButtonLink(link: INavLink, linkIndex: number) {
-    return (
+  private _renderButtonLink(link: INavLink, linkIndex: number, nestingLevel: number) {
+    const isRtl: boolean = getRTL();
+    const paddingBefore = _indentationSize * nestingLevel +
+      (this._hasExpandButton ? _indentWithExpandButton : _indentNoExpandButton);
+
+      return (
       <CommandButton
         className={ css(
           'ms-Nav-link ms-Nav-linkButton',
           styles.link,
-          this._hasExpandButton && 'isOnExpanded ' + styles.linkIsOnExpanded
+          this._hasExpandButton && 'isOnExpanded ' + styles.linkIsOnExpanded,
+          !link.icon && styles.commandButtonNoIcon
         ) }
+        style={ { [isRtl ? 'paddingRight' : 'paddingLeft']: paddingBefore + 'px'} }
```

Comment:
style tag is soon going to be a linting error. we're moving towards using mergeStyles for all CSS declared in JS. Here's an example.

https://github.com/OfficeDev/office-ui-fabric-react/blob/9a5a9e2f3583ffda8e60a155f96c800e28a9af47/packages/experiments/src/components/LayoutGroup/LayoutGroup.tsx#L30

It can take a list of class names, or CSS objects.

Reply:
I have tried to follow your suggestion, by changing the style as follows:
```
<CommandButton
  className={
    mergeStyles(
      'ms-Nav-link ms-Nav-linkButton',
      styles.link,
      this._hasExpandButton && 'isOnExpanded ' + styles.linkIsOnExpanded,
      !link.icon && styles.commandButtonNoIcon,
      isRtl && {
        paddingRight: paddingBeforePixels
      },
      !isRtl && {
        paddingLeft: paddingBeforePixels
      }
    ) as string
  }
. . .
```
However, it looks like the `ms-Nav-Link` class already has a definition for `padding-left`: this definition was overridden by the `styles` prop passed to `CommandButton`, but passing the CSS object to mergeStyles does not have the same effect.

Do you have any thoughts on how to fix this?

( appending `!important` to `paddingBeforePixels` fixes the issue, but I don't feel that's the right solution)

Reply:
Can you move the padding from ms-Nav-Link into this mergeStyles block? Under what circumstances is that padding-left needed?

Reply:
Looking more carefully, I noticed that the `.link` class was responsible for that padding.

I have removed all padding definitions from the CSS, and let the TS code + `mergeStyles` handle it now. I still had to add an `!important` for the `CommandButton` types, because some internal styles were overriding the styles passed in.